src/View/Helper/ListsHelper.php

```diff
@@ -535,33 +578,45 @@ public function displayAddSentenceForm($listId)
     }
 
 
-    public function displayListsModule($listsArray)
+    public function displayListsModule($allListsArray, $sentences)
     {
-        if (count($listsArray) > 0) {
+        $sentenceId = (int)$sentences->id; 
+        
+        if (count($allListsArray) > 0) {
             echo '<div class="section md-whiteframe-1dp">';
             /* @translators: header text on the sidebar of a sentence page */
             echo $this->Html->tag('h2', __('Lists'));
             echo '<ul class="sentence-lists">';
-            foreach($listsArray as $list) {
+            foreach($allListsArray as $list) {
                 $list = $list->sentences_list;
-                if ($list['visibility'] == 'public') {
+                if ($list['visibility'] == 'public' || $list['visibility'] == 'public+') {
                     $class = 'public-list';
                 } else {
                     $class = 'personal-list';
                 }
                 echo '<li class="'.$class.'">';
-                echo $this->Html->link(
+                 echo $this->Html->link(
                     $this->_View->safeForAngular($list['name']),
                     array(
                         'controller' => 'sentences_lists',
                         'action' => 'show',
                         $list['id']
                     )
                 );
+                 if(CurrentUser::isMember())
+                 {
+                    echo $this->_displayRemoveLink($list['id'], $sentenceId, $list['name']);
+                 }
```

Comment:
As written, the remove button is displayed after each list even if the current user is not the list owner.

You should only display the button if the list belongs to the current user.
Currently you don't have this information available but if you modify `SentencesSentencesListsTable::getOwnAndPublicListsForSentence()` to include the list owner's id, the correct condition would be `CurrentUser::get('id') == $list->user_id`.